### PR TITLE
VB-2362 front end delete a session template

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
Changes to allow session template deletion.

### CODE REVIEW

**Review by tpmcgowan — COMMENTED** [review]
Looks great 👍. Just a minor suggestion about making the route for delete have a more explicit path.

File: server/routes/prisons/sessionTemplates/index.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ export default function routes(services: Services): Router {
   get('/prisons/:prisonId([A-Z]{3})/session-templates/:reference', singleSessionTemplate.view())
   post('/prisons/:prisonId([A-Z]{3})/session-templates/:reference/activate', singleSessionTemplate.activate())
   post('/prisons/:prisonId([A-Z]{3})/session-templates/:reference/deactivate', singleSessionTemplate.deactivate())
+  post('/prisons/:prisonId/session-templates/:reference', singleSessionTemplate.delete())
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply server/routes/prisons/sessionTemplates/index.ts]: Updated to add /delete and refactored tests based on this change.

File: server/routes/prisons/sessionTemplates/singleSessionTemplateController.ts [hunk 2]
```diff
@@ -62,4 +62,21 @@ export default class SingleSessionTemplateController {
...
+
+  public delete(): RequestHandler {
+    return async (req, res) => {
+      const { reference, prisonId } = req.params
+
+      try {
+        await this.sessionTemplateService.deleteSessionTemplate(res.locals.user.username, reference)
+        req.flash('message', `Session template with reference ${reference} deleted.`)
+      } catch (error) {
+        req.flash('formValues', req.body)
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply server/routes/prisons/sessionTemplates/singleSessionTemplateController.ts]: Updated

**Review by tpmcgowan — APPROVED** [review]
LGTM!

**Pull request merged**